lms/djangoapps/courseware/models.py

```diff
@@ -400,3 +400,34 @@ class Meta(object):
         default=False,
         help_text=_('Disable the dynamic upgrade deadline for this course run.')
     )
+
+
+class CourseScheduleConfiguration(ConfigurationModel):
+    """
+    Per-course run configuration for dynamic upgrade deadlines.
+
+    This model controls dynamic upgrade deadlines on a per-course run level, allowing course runs to
+    have different deadlines or opt out of the functionality altogether.
+    """
+    class Meta(object):
+        app_label = 'courseware'
+
+    KEY_FIELDS = ('course_id',)
+
+    course_id = CourseKeyField(max_length=255, db_index=True)
+    verified_upgrade_deadline_days = models.PositiveSmallIntegerField(
```

Comment:
How does this differ from `CourseDynamicUpgradeDeadlineConfiguration.deadline_days`?